lib/matplotlib/lines.py

```diff
@@ -790,6 +790,9 @@ def draw(self, renderer):
                     cap = self._dashcapstyle
                     join = self._dashjoinstyle
                 else:
+                    print(tpath.vertices.size)
+                    tpath = affine.inverted().transform_path(affine.transform_path(tpath).cleaned(simplify=True))
+                    print(tpath.vertices.size)
```

Comment:
I guess what is happening is that you need to do the simplification in screen units, but the later draw routine needs the path in data units, so you need to convert one extra time back and forth?  That seems inefficient, any chance you could either 1. do the simplification at the last minute (once you don't need to convert back and forth anymore), or at least 2. keep the simplified path in screen units, and set the transform to identity?

Also, instead of prints (which need to go anyways :-)), some tests (e.g. with paths for which you know what the result of simplification should be) would be welcome, to show that the simplifier is indeed working as expected.